src/pset/map/input.rs

```diff
@@ -258,6 +260,95 @@ pub struct Input {
     pub unknown: BTreeMap<raw::Key, Vec<u8>>,
 }
 
+/// A Signature hash type for the corresponding input. As of taproot upgrade, the signature hash
+/// type can be either [`SigHashType`] or [`SchnorrSigHashType`] but it is not possible to know
+/// directly which signature hash type the user is dealing with. Therefore, the user is responsible
+/// for converting to/from [`PsbtSighashType`] from/to the desired signature hash type they need.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct PsbtSighashType {
+    pub (crate) inner: u32,
```

Comment:
Specific reason why here you do a `pub (crate) inner: u32` and for the `NonStandardSighashType`, you just have a newtype with `pub u32`?

Reply:
No. Just an artifact copied from rust-bitcoin.